apiserver/routers.go

```diff
@@ -41,5 +42,34 @@ func (api *APIServer) NewRouter(ctx context.Context) *mux.Router {
 		Name("GetDistroSigImageConfig").
 		HandlerFunc(api.GetDistroSigImageConfig)
 
+	router.Methods("GET").Path("/healthz").Name("healthz").HandlerFunc(healthz)
+
+	// global timeout and panic handlers.
+	router.Use(timeoutHandler(), recoveryHandler())
+
 	return router
 }
+
+func healthz(w http.ResponseWriter, r *http.Request) {
+	handleOK(w, r)
+}
+
+func setHeaders(w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+}
+
+func handleOK(w http.ResponseWriter, r *http.Request) {
+	setHeaders(w)
+	w.WriteHeader(http.StatusOK)
+}
+
+func recoveryHandler() mux.MiddlewareFunc {
+	return handlers.RecoveryHandler(handlers.PrintRecoveryStack(true))
+}
+
+func timeoutHandler() mux.MiddlewareFunc {
```

Comment:
this should behavior very similarly to our timeout goroutine, but properly tracks recovery by passing panic stacks across goroutines, which is very tricky. our implementation was not close to cutting it. https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/net/http/server.go;l=3394-3437